File: csharp/Src/Shapes/Base/Commons/Displayable.cs
namespace Shapes.Base.Commons;

public interface IDisplayable {
    bool ShouldDisplay() {
        return true;
    }

    void Display() {
        if (ShouldDisplay()) {
            Console.WriteLine("Yes, this shape can be displayed.");
        } else {
            Console.WriteLine("No, this shape cannot be displayed.");
        }
    }
}

File: elixir/lib/shapes/base/commons/displayable.ex
alias Shapes.Base.Commons.Displayable

defmodule Displayable do
  @spec should_display(struct()) :: boolean()
  def should_display(_shape) do
    true
  end

  @spec display(struct()) :: :ok
  def display(shape) do
    if should_display(shape) do
      IO.puts("Yes, this shape can be displayed.")
    else
      IO.puts("No, this shape cannot be displayed.")
    end
  end
end
